lib/index.js

```diff
@@ -290,3 +299,74 @@ internals.stop = async (server) => {
 
 internals.state = (realm) => Toys.state(realm, 'schwifty');
 internals.rootState = (realm) => Toys.rootState(realm, 'schwifty');
+
+internals.migrateDiffCmd = async (srv, args, rootDir, ctx) => {
+
+    const SchwiftyMigrateDiff = SchwiftyMigrateDiffHelper.get();
+
+    if (!SchwiftyMigrateDiff) {
+        throw new ctx.DisplayError('To use this command please first install `npm install --save-dev schwifty-migrate-diff`');
+    }
+
+    const definition = {
+        mode: {
+            description: 'Set mode to "create" or "alter" (defaults to "alter") -- when set to create the migration file ignores changes that will delete or mutate existing columns/tables'
+        },
+        dir: {
+            description: 'Specifies the directory in which to write the migration file, defaults to the migrationsDir Schwifty receives at registration'
+        }
+    };
+
+    const parsedArgs = Bossy.parse(definition, { argv: args });
+
+    if (parsedArgs instanceof Error) {
+        throw new ctx.DisplayError(parsedArgs.message + Os.EOL + Bossy.usage(definition));
+    }
+
+    Hoek.assert(parsedArgs._ && parsedArgs._.length !== 0, new ctx.DisplayError('A name must be specified for the generated migration'));
+    Hoek.assert(parsedArgs._.length === 1, new ctx.DisplayError(Bossy.usage(definition)));
+
+    const [migrationName] = parsedArgs._;
+
+    if (parsedArgs.mode) {
+        Hoek.assert(parsedArgs.mode === 'create' || parsedArgs.mode === 'alter', new ctx.DisplayError(Bossy.usage(definition)));
+    }
+
+    const { realmsWithMigrationsDir } = internals.state(srv.realm).collector;
+
+    const realmsWithMigrationsInRoot = Array.from(realmsWithMigrationsDir).reduce((collector, realm) => {
+
+        const { migrationsDir } = internals.state(realm);
+        if (internals.isChildPath(migrationsDir, rootDir)) {
+            collector.push(realm);
+        }
+
+        return collector;
+    }, []);
+
+    Hoek.assert(realmsWithMigrationsInRoot.length > 0, new ctx.DisplayError(`At least 1 migrations directory must be present inside ${rootDir}`));
+    Hoek.assert(realmsWithMigrationsInRoot.length === 1, new ctx.DisplayError(`Only 1 migrations directory supported per project at this time. Please remove one of [${realmsWithMigrationsInRoot.map((realm) => realm.plugins.schwifty.migrationsDir)}]`));
+
+    const [migrationRealm] = realmsWithMigrationsInRoot;
+
+    const serverModels = srv.models();
+    const allModels = Object.keys(serverModels).map((modelName) => serverModels[modelName]);
+    const realmKnex = internals.state(migrationRealm).knex;
+    const models = allModels.filter((model) => model.knex() === realmKnex);
+    const modelsArr = Object.keys(models).map((key) => models[key]);
```

Comment:
Pretty sure this doesn't do anything but clone `models`, which is already an array!

Reply:
Oh! Super weird that this doesn't break everything, seeing as how `models` there is being treated like an object

Reply:
```js
> Object.keys([1,2,3])
[ '0', '1', '2' ]
```

;)